src/workload.py

```diff
@@ -85,13 +84,9 @@ def exec(self, command: list[str], working_dir: str | None = None) -> str:
             cwd=working_dir,
         )
 
+    @property
     @override
-    @retry(
-        wait=wait_fixed(1),
-        stop=stop_after_attempt(5),
-        retry_error_callback=lambda state: state.outcome.result(),  # type: ignore
```

Comment:
**todo:** When we remove this, if we fail the retry conditions, we'll raise a `RetryError`, which we're definitely not handling in the charm.

This logic returns the default 'False', which I think is better.

Reply:
Yes, the issue I found on these retries, is that we are already handling the exception within the code itself by returning, we don't let the exceptions raise. So that logic on itself already makes the retry obsolete, other than to guard around random exceptions, and I don't think we should be making defensive code against _any_ exception here. Better to be explicit imho.

Maybe the `retry=retry_if_result(lambda result: result is False)` clause can be kept, but I'm not sure I see the use of `retry_error_callback`.

There is another issue with the callback as it was, which is that it won't return a `True | False` but an Object, making the alive check pass because of _truthy_ behaviour.

Reply:
After a thought, maybe we can have this?
```python
@retry(
    wait=wait_fixed(1),
    stop=stop_after_attempt(5),
    retry=retry_if_result(lambda result: result is False),
    retry_error_callback=lambda state: False,
)
```